tubemq-manager/src/main/java/org/apache/tubemq/manager/service/NodeService.java

```diff
@@ -413,4 +415,32 @@ public AddBrokerResult addBrokersToClusterWithId(AddBrokersReq req, NodeEntry ma
         }
         return null;
     }
+
+    public TubeMQResult cloneTopicToBrokers(CloneTopicReq req, NodeEntry master) throws Exception {
+
+        // 1 query topic config
+        TubeHttpTopicInfoList topicInfoList = requestTopicConfigInfo(master, req.getSourceTopicName());
+
+        if (topicInfoList == null) {
+            return TubeMQResult.getErrorResult("no such topic");
+        }
+
+        // 2 if there's no specific broker ids then clone to all of the brokers
+        List<Integer> brokerId = req.getBrokerId();
+
+        if (CollectionUtils.isEmpty(brokerId)) {
+            TubeHttpBrokerInfoList brokerInfoList = requestClusterNodeStatus(master);
+            if (brokerInfoList != null) {
+                brokerId = brokerInfoList.getAllBrokerIdList();
```

Comment:
make sure all brokers are in configurable status
TubeHttpBrokerInfoList.getConfigurableBrokerIdList

Reply:
done